kedro/framework/cli/utils.py

```diff
@@ -401,11 +397,41 @@ def _config_file_callback(ctx, param, value):  # noqa: unused-argument
 
     if value:
         config = OmegaConf.to_container(OmegaConf.load(value))[section]
+        for key, value in config.items():
+            _validate_config_file(key)
         ctx.default_map.update(config)
 
     return value
 
 
+def _validate_config_file(key):
+    """Validate the keys provided in the config file against the accepted keys."""
+    accepted_keys = [
+        "from_inputs",
+        "to_outputs",
+        "from_nodes",
+        "to_nodes",
+        "node_names",
+        "runner",
+        "env",
+        "tags",
+        "load_versions",
+        "pipeline",
+        "namespace",
+        "conf_source",
+        "params",
+        "is_async",
+        "tag",
+        "nodes_names",
+        "load_version",
```

Comment:
```
from kedro.framework.cli.project import run

# Inspect arguments
args = [click_arg.name for click_arg in run.params]
print(args)
```

(need to remove `config` from the list)

I played around with it you can retrieve the argument list of a click command.

Reply:
Omg amazing!! I tried to find how to do this yesterday, but struggled because it's a click function. Thank you 😄